api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -116,6 +117,25 @@ module.exports = {
       });
   },
 
+  getByUsernameOrEmailWithRoles(username) {
+    return BookshelfUser
+      .query((qb) => { qb.whereRaw('LOWER("email") LIKE ?', `${username.toLowerCase()}`).orWhere({ 'username': username }); })
+      .fetch({
+        withRelated: [
+          'memberships',
+          'memberships.organization',
+          'pixRoles',
+          'certificationCenterMemberships.certificationCenter',
+        ]
+      })
+      .then((foundUser) => {
+        if (foundUser === null) {
```

Comment:
Est-ce qu'il ne vaut pas mieux être plus "large" ici ? 
```js
if(!foundUser) 
```

Reply:
Le get renvoie une instance unique car les deux colonnes email et username ont une colonne d'unicité. Cela couvre notre cas par contre je vois deux manière de gérer dans notre code soit un test via un null puis un reject lors du not found ou on catch l'exception renvoyer par le get et on la rethrow. Je serai plus d'avis à garder comme cela car c'est du code ancien. Neanmois pour enrichir ce thread https://1024pix.atlassian.net/wiki/spaces/DEV/pages/561119373/2018-11-22+Enqu+te+nomenclature+des+getXXX+et+findXXX+de+Repositories choisir la manière de gérer les notfound dans le repository voir s'il y a une préference histoire d'homgéniser